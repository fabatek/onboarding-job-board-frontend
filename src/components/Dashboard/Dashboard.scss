@import '../AllJobs/AllJobs';
.dashBoard{
    
    .search{
        .search__form{
            display: flex;
            align-items: center;
            justify-content: center;
        }
    }
    .paginationBttns {
        width: 80%;
        height: 40px;
        list-style: none;
        display: flex;
        justify-content: center;
        align-items: flex-end;
        margin-top: 20px;
         
        a {
            padding: 10px;
            margin: 8px;
            border-radius: 5px;
            border: 1px solid #2b2eff;
            color: #2b2eff;
            cursor: pointer;
            &:hover {
                color: white;
                background-color: #2b2eff;
              }
          }
          
      }
      .paginationActive a {
        color: white;
        background-color: #2b2eff;
      }
      
      .paginationDisabled a {
        color: grey;
        background-color: grey;
      }
}
