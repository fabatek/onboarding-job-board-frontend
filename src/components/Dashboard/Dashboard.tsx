import React,{useState, useEffect, FC} from 'react'
import './Dashboard.scss'
import { Dropdown, Form } from 'react-bootstrap';
import ReactPaginate from 'react-paginate';
import { JobTotal } from '../TotalJobs/Job';
import { useSelector, useDispatch } from 'react-redux';
import {fetchJobs} from '../../data/api'
interface RootState {
    Job: [],
    loading: boolean,
    error: null
  }
const Dashboard : FC = () => {
    const [pageNumber, setPageNumber] = useState(0);
    const jobsPerPage = 10; 
    const pagesVisited = pageNumber * jobsPerPage;
    const [searchTerm,setSearchTerm ] = useState("");
    const jobs = useSelector((state: RootState) => state.Job.slice(0, 100));
    const dispatch = useDispatch()
     useEffect(() => {
        dispatch(fetchJobs())
        }, [dispatch])
        
    //Pagination
    const pageCount = Math.ceil(jobs.length / jobsPerPage);
    const changePage = ( {selected}:any ) => {
            setPageNumber(selected);
        };
    
    return (
        <div className="dashBoard">
            <JobTotal/>
            <div className='search'>
                <div className="search__form">
                    
                    <Form.Control type="text" placeholder="Search..." onChange={(e)=>{ setSearchTerm(e.target.value)}} />
                    
                    <Dropdown>
                        <Dropdown.Toggle variant="secondary" id="dropdown-basic">
                            City
                        </Dropdown.Toggle>

                        <Dropdown.Menu>
                            <Dropdown.Item href="#/action-1">Ha Noi</Dropdown.Item>
                            <Dropdown.Item href="#/action-2">Ho Chi Minh</Dropdown.Item>
                            <Dropdown.Item href="#/action-3">Da Nang</Dropdown.Item>
                        </Dropdown.Menu>
                    </Dropdown>
                </div>
            </div>
                {jobs.filter((val:any)=>{
                    if(searchTerm===""){
                        return val;
                    } else if(
                        val.jobName.toLowerCase().includes(searchTerm.toLowerCase()) 
                    ){
                        return val;
                    }
                    
                    })
                    .slice(pagesVisited, pagesVisited + jobsPerPage)
                    .map((jobs:any) => {
                        return (
                            <div key={jobs.id} className="jobsCard">
                                <div className="jobsCard__left">
                                    <img src={jobs.jobImg} alt="" />
                                </div>
                                <div className="jobsCard__center">
                                    <a href="" className="name">{jobs.jobName}</a>
                                    <a href="" className="company">{jobs.jobCompany}</a>
                                    <a href="" className="type">{jobs.jobType}</a>
                                </div>
                                <div className="jobsCard__right">
                                    <p>{jobs.jobArea}</p>
                                </div>
                                
                            </div>

                        );
                    })}
                    <ReactPaginate 
                    previousLabel={"Previous"}
                    nextLabel={"Next"}
                    pageCount={pageCount}
                    onPageChange={changePage}
                    containerClassName={"paginationBttns"}
                    previousLinkClassName={"previousBttn"}
                    nextLinkClassName={"nextBttn"}
                    disabledClassName={"paginationDisabled"}
                    activeClassName={"paginationActive"}
                    pageRangeDisplayed={10} marginPagesDisplayed={10}
                    data-testid="pagination"/>
        </div> 
    )
}

export default Dashboard
