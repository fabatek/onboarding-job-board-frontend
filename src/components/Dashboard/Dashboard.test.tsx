import {render, fireEvent, screen} from "@testing-library/react";
import Dashboard from "./Dashboard";
import React from "react";
import ReactPaginate from "react-paginate";
describe("Step form reandering", () => {
  it('it should render a step form`', () => {
    const {container} = render(<Dashboard />);
    
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
describe('<Dashboard />', () => {
    it('when rendering success', () => {
      const {container} = render(<Dashboard />);
      expect(container.querySelectorAll('.dashBoard').length).toBe(1);
    });
});
