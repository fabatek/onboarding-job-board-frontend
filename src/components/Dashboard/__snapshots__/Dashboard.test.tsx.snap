// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step form reandering it should render a step form\` 1`] = `
<div>
  <div
    class="dashBoard"
  >
    <h1>
      Có tất cả 
      0
       IT Jobs For Chất Developers
    </h1>
    <div
      class="search"
    >
      <div
        class="search__form"
      >
        <form
          autocomplete="off"
          class="MuiBox-root css-1eucw4r"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Search
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    Search
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </form>
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-168o85v-MuiAutocomplete-root"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="combo-box-demo"
              id="combo-box-demo-label"
            >
              City
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="combo-box-demo"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    City
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="paginationBttns"
    >
      <li
        class="previous paginationDisabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          class="previousBttn"
          rel="prev"
          role="button"
          tabindex="0"
        >
          Previous
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          class="nextBttn"
          rel="next"
          role="button"
          tabindex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;
